File: 2432-number-of-zero-filled-subarrays/number-of-zero-filled-subarrays.java
class Solution {
    public long zeroFilledSubarray(int[] nums) {
        long res = 0;
        int l = -1;
        for (int r=0; r<nums.length; ++r) {
            if (nums[r] == 0) res += (r-l);
            else l = r;
        }
        return res;
    }
}

File: 2432-number-of-zero-filled-subarrays/number-of-zero-filled-subarrays.py
class Solution:
    def zeroFilledSubarray(self, nums: List[int]) -> int:
        res = 0
        l = -1
        for r in range(len(nums)):
            if nums[r] == 0:
                res += (r - l)
            else:
                l = r

        return res